139 filling the gel rig with the gel

tylerhand1 opened the pull request:
In addition to fixing issues, I added a new group, "Ionic Substance," as a check for the electrolysis setup to ensure that it can only be filled with substances with this type, instead of all liquid substances.

jcourt325 commented on project/source/Scripts/Objects/ElectrolysisSetup.gd:
```diff
@@ -104,6 +108,8 @@ func slot_filled(slot, object):
 		slot_emptied(slot, object)
 
 func slot_emptied(slot, object):
+	if "Fillable" in object:
```
It looks to me like this change is intended to reset the Gel Mold to be fillable again after being removed from the slot.

If that's true, I think we should, instead of using `in object`, check if the object `is_in_group('Gel Boat')` (like you do above in `slot_filled`). Because by checking it this way, we don't actually know for sure whether it is a gel boat, or whether it's some other mystery object that also has a `Fillable` property, which it could theoretically use in a different way and expect not to be modified like this. And if nothing else I also think this would make the code a little more readable.

jcourt325 commented on project/source/Scripts/Objects/WeighBoat.gd:
```diff
@@ -8,7 +8,7 @@ func _ready():
 	pass
 
-func AddContents(new_contents):
+func AddContents(new_contents) -> bool:
```
Do we now need to put something in the Groups documentation table that specifies this should return a `bool`?

tylerhand1 replied:
I don't believe so, if we opt for the change I agreed with above regarding the `active` property. Instead, we should document that on the `LabObject` page.

If we opt for that change, I think we should revert this back.

jcourt325 reviewed (changes requested):
I think this mostly looks good, but I did have a few questions/concerns that I wanted to talk about before merging this. I've added comments on the individual changes for those.

This feedback is based entirely on reading the Files Changed, so tell me if I'm wrong about anything and why.

jcourt325 commented on project/source/Scenes/Substances/GelModule/BinderSubstance.tscn:
```diff
@@ -2,5 +2,5 @@
 [ext_resource path="res://Scenes/Substances/GelModule/BinderSubstance.gd" type="Script" id=1]
 
-[node name="BinderSubstance" type="Node2D" groups=["Chillable", "Conductive", "Heatable", "Liquid Substance", "Mixable", "Weighable"]]
+[node name="BinderSubstance" type="Node2D" groups=["Chillable", "Conductive", "Heatable", "Ionic Substance", "Liquid Substance", "Mixable", "Weighable"]]
```
Also, do does "Ionic Substance" need to be added to the Groups documentation?

tylerhand1 replied:
Yes, once the changes for this branch are finalized and merged into dev, I can create an issue for it and I could add that to the Groups documentation.

tylerhand1 commented on project/source/Scripts/Objects/GelMoldSubsceneManager.gd:
```diff
@@ -18,6 +18,10 @@ var contents = []
 
+# This determines whether it can be filled,
+# so that it can prevent taking substances when another object needs that substance
+var Fillable: bool = true
```
I agree that I lean towards the former where we have that general property of `active` or something along those lines.

If we opt for that change, I think it would be best to set this `active` property within `ObjectSlot.gd` within the `TryInteract()` and `_on_GelBoatSlot_input_event` functions.

tylerhand1 commented:
I have made changes that switched to the more generic approach of using `active` instead of `Fillable` for determining whether an object should interact.

I have also removed the need for the check of 'if "Fillable in object' within `ElectrolysisSetup.gd`, so that issue is no longer a problem.

tylerhand1 commented:
I deleted the edited `GelImager.gd` file and placed my changes in the proper file and fixed the file path to point to this `GelImager.gd`. I feel this is a necessary change to make the folder structure more organized.

I also made a small change to fix the UI menu for the `ElectrolysisSetup` fill prompt.
